### PR TITLE
Internal 280 cypress auth

### PR DESCRIPTION
**Pull request by andrea-grapl**
### Which issue does this PR correspond to?

https://github.com/grapl-security/issue-tracker/issues/280

### What changes does this PR make to Grapl? Why?
Adds testing for authentication because we previously didn't have a tests to validate that authentication is working appropriately. 

### How were these changes tested?
Using Cypress locally:

- Run `make up`
- `cd test` from the root of grapl
- Change baseUrl from `"baseUrl": "http://api.grapl.test:3128",` `cypress.json` to `baseUrl: http://localhost:1234,`
- Run `npx cypress open`
- Click `login.js` // cypress tests will run

Change URL back to `http://api.grapl.test:3128" & validated Grapl Build integration tests checks pass when submitting PR.

Reference: https://stackoverflow.com/questions/60031340/cypress-login-command-with-cy-request-causes-subsequent-cy-visit-to-fail

### CODE REVIEW

File: test/cypress/integration/login.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+describe("sanity check", () => {
```
Comment (wimax-grapl) [inline test/cypress/integration/login.js:1]: probably not needed

File: test/cypress/integration/login.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	it("passes", () => {
+		expect(true).to.equal(true);
+	});
+});
+
```
Comment (wimax-grapl) [inline test/cypress/integration/login.js:6]: could we replace all the tabs here with four spaces, if possible?

File: test/cypress.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
     "baseUrl": "http://api.grapl.test:3128",
-    "pageLoadTimeout": 180000
+    "pageLoadTimeout": 180000, 
+    "chromeWebSecurity": false
```
Comment (wimax-grapl) [inline test/cypress.json:4]: is this line absolutely needed?

[the remark this answers is not in the document]
Comment (ajm-grapl) [reply test/cypress.json:4]: I was able to get passes with `make test-integration` without this line, FWIW.

File: test/cypress/integration/login.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+			const grapl_jwt = { user: { authenticationData: { token: body.token } } };
+			window.localStorage.setItem("grapl_jwt", JSON.stringify(grapl_jwt));
+		});
+		cy.contains("login").should("not.exist");
+	});
+});
+
+describe("checks that cookie was set after login", () => {
+	it("retrieves grapl_jwt", () => {
+		cy.getCookie("grapl_jwt");
```
Comment (ajm-grapl) [inline test/cypress/integration/login.js:50]: I tried this as `cy.getCookie("grapl_jwt").should('exist')` and got a failure, saying the cookie was null.

AFAICT the Cypress API will only fail for unexpected conditions if you use an assertion method, like "should".

**Review by ajm-grapl — CHANGES_REQUESTED** [review]
An explicit assertion on the jwt token causes the test to fail

File: test/cypress.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrea-grapl) [reply test/cypress.json:4]: It's for CORS

**Pull request closed without merging**